src/aggregates/follows.ts

```diff
@@ -6,12 +6,24 @@ import {
   UnsequencedChannelEvent,
 } from '../models/ChannelEvent'
 
+import { ChannelFollowsInfo } from '../entities/ChannelFollowsInfo'
+
 type ChannelEventsAggregationResult = {
   events?: ChannelEvent[]
 }[]
 
 export class FollowsAggregate implements GenericAggregate<ChannelEvent> {
   private channelFollowsMap: Record<string, number> = {}
+  private allChannelFollowsEvents: Partial<UnsequencedChannelEvent>[] = []
+  private allChannelFollows: ChannelFollowsInfo[] = []
+
+  private addOrUpdateFollows(array: ChannelFollowsInfo[], id: string, event: ChannelEventType): void {
+    const i = array.findIndex((element) => element.id === id)
+    if (i > -1) {
+      if (!array[i].follows && event === ChannelEventType.UnfollowChannel) return
+      array[i].follows = event === ChannelEventType.FollowChannel ? array[i].follows + 1 : array[i].follows - 1
```

Comment:
You need to make sure to not get to negative values

Reply:
But there is a check for that in line 23

Reply:
Okay, missed that